### PR TITLE
Make vue a peer dependency

### PR DESCRIPTION
**Pull request by znck**
## What I did
Depend loosely on vue & vue-loader by moving them to peer dependencies. This allows end user to
pick any version of vue for their project.

## How to test
Existing tests should pass.

### CODE REVIEW

**Review by kazupon — APPROVED** [review]
Thanks!

**Comment by Hypnosphi** [comment]
@kazupon Is there a way we can merge this PR with #2032?

**Comment by kazupon** [comment]
@Hypnosphi Once merge #2032 to `master` branch, after that pull from this `master` branch to this PR by [Update branch] of GitHub's WebUI. How is a way?

**Comment by Hypnosphi** [comment]
I mean, that one needs review, but it partially interleaves with this one, so I'm not sure what should we do

**Comment by Hypnosphi** [comment]
Btw, can we confirm that we work with **any** vue version?

File: app/vue/package.json [hunk 1]
```diff
@@ -70,17 +70,22 @@
...
   "devDependencies": {
     "babel-cli": "^6.26.0",
-    "nodemon": "^1.12.1"
+    "nodemon": "^1.12.1",
+    "vue": "^2.5.2",
+    "vue-loader": "^13.3.0",
+    "vue-template-compiler": "^2.5.2",
+  },
+  "peerDependencies": {
+    "vue": "*",
```
Comment (Hypnosphi) [inline app/vue/package.json]: Can we check that we work with **any** vue version?

File: app/vue/package.json [hunk 2]
```diff
@@ -70,17 +70,22 @@
...
     "webpack-dev-middleware": "^1.12.0",
     "webpack-hot-middleware": "^2.19.1"
   },
   "devDependencies": {
     "babel-cli": "^6.26.0",
-    "nodemon": "^1.12.1"
+    "nodemon": "^1.12.1",
+    "vue": "^2.5.2",
+    "vue-loader": "^13.3.0",
+    "vue-template-compiler": "^2.5.2",
```
Comment (Hypnosphi) [inline app/vue/package.json]: Trailing comma makes JSON invalid

**Review by Hypnosphi — CHANGES_REQUESTED** [review]

File: app/vue/package.json (hunk 1 above)
Comment (kazupon) [reply app/vue/package.json]: I think that we should be checked vue latest version only. because `vue-template-compiler` forces the same version as `vue`

Comment (Hypnosphi) [reply app/vue/package.json]: I mean, do we work with vue version 1? If not, we shouldn't put an asterisk here

File: app/vue/package.json [hunk 3]
```diff
@@ -70,17 +70,22 @@
     "url-loader": "^0.6.2",
     "util-deprecate": "^1.0.2",
     "uuid": "^3.1.0",
-    "vue": "^2.5.2",
-    "vue-hot-reload-api": "^2.2.0",
-    "vue-loader": "^12.2.1",
+    "vue-hot-reload-api": "^2.1.1",
```
Comment (Hypnosphi) [inline app/vue/package.json]: It should be `2.2.0`

**Review by Hypnosphi — CHANGES_REQUESTED** [review]

File: app/vue/package.json [hunk 4]
```diff
@@ -70,17 +70,22 @@
...
     "webpack": "^3.6.0",
     "webpack-dev-middleware": "^1.12.0",
     "webpack-hot-middleware": "^2.20.0"
   },
   "devDependencies": {
     "babel-cli": "^6.26.0",
-    "nodemon": "^1.12.1"
+    "nodemon": "^1.12.1",
+    "vue": "^2.5.2",
+    "vue-loader": "^13.3.0",
```
Comment (Hypnosphi) [inline app/vue/package.json]: What are the breaking changes between 12 and 13?

File: app/vue/package.json (hunk 3 above)
Comment (Hypnosphi) [reply app/vue/package.json]: Please run `yarn install` to update lockfile after you fix it

File: app/vue/package.json (hunk 4 above)
Comment (eriknygren) [reply app/vue/package.json]: @Hypnosphi https://github.com/vuejs/vue-loader/releases/tag/v13.0.0

Upgrading `vue-loader` also fixes https://github.com/vuejs/vue-loader/issues/951 (supporting nested css rules). Which is something I use heavily in my .vue files and that's currently blocking me from adding several components to my storybook.

So would love to see this getting merged eventually!

File: app/vue/package.json (hunk 3 above)
Comment (ndelangen) [reply app/vue/package.json]: Not a deal-breaking for this PR, TBH.

File: app/vue/package.json (hunk 4 above)
Comment (ndelangen) [reply app/vue/package.json]: ✅

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
@Hypnosphi Looks like you have no huge objections to this PR, so I hope you don't mind me merging this as Vue-version mismatches are a huge problem for users.

**Comment by ndelangen** [comment]
Thank you for your efforts @znck @kazupon @eriknygren @Hypnosphi 👍 💯

**Comment by Hypnosphi** [comment]
Actually, I had one: https://github.com/storybooks/storybook/pull/2041#discussion_r145095359

**Pull request merged**